system-addon/content-src/components/Card/Card.jsx

```diff
@@ -0,0 +1,50 @@
+const React = require("react");
+const LinkMenu = require("content-src/components/LinkMenu/LinkMenu");
+const shortURL = require("content-src/lib/short-url");
+const {FormattedMessage} = require("react-intl");
+const cardContextTypes = require("./types");
+
+class Card extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {showContextMenu: false, activeCard: null};
+  }
+  toggleContextMenu(event, index) {
+    this.setState({showContextMenu: true, activeCard: index});
+  }
+  render() {
+    const {index, link} = this.props;
+    const isContextMenuOpen = this.state.showContextMenu && this.state.activeCard === index;
+    const hostname = shortURL(link);
+    const {icon, intlID} = cardContextTypes[link.type];
+
+    return (<li className={`card${isContextMenuOpen ? " active" : ""}`} tabIndex="1">
+      <a href={link.url}>
+        {link.image && <div className="card-preview-image"><img src={link.image} /></div>}
+        <div className="card-details">
+          <div className="card-host-name"> {hostname} </div>
+          <h4 className="card-title"> {link.title} </h4>
+          <p className={`card-description${link.image ? "" : " full-height"}`}> {link.description} </p>
+          <div className="card-context">
+            <span className={`card-context-icon icon icon-${icon}`} />
+            <div className="card-context-label"><FormattedMessage id={intlID} defaultMessage="Visited" /></div>
+          </div>
+        </div>
+      </a>
+      <button className="context-menu-button" tabIndex="0"
+        onClick={e => {
+          e.preventDefault();
+          this.toggleContextMenu(e, index);
+        }}>
+        <span className="sr-only">{`Open context menu for ${link.title}`}</span>
+      </button>
+      <LinkMenu
```

Comment:
This is expecting a `dispatch` prop, so you need to either refactor` LinkMenu` to be connected or somehow pass it in here.